src/Servers/Kestrel/Core/src/Internal/WebTransport/WebTransportSession.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Net.Http;
+using System.Threading.Channels;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Features;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http3;
+using Microsoft.AspNetCore.Server.Kestrel.Core.WebTransport;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.WebTransport;
+
+/// <summary>
+/// Controls the WebTransport session and keeps track of all the streams.
+/// </summary>
+internal class WebTransportSession : IWebTransportSession
+{
+    private readonly ConcurrentDictionary<long, WebTransportStream> _openStreams = new();
+    private readonly Channel<WebTransportStream> _pendingStreams;
+    private readonly Http3Connection _connection;
+    private readonly Http3Stream _connectStream = default!;
+    private bool _isClosing;
+
+    private static readonly ReadOnlyMemory<byte> _outputStreamHeader = new(new byte[] {
```

Comment:
Static use a different style.
```suggestion
    private static readonly ReadOnlyMemory<byte> OutputStreamHeader = new(new byte[] {
```